File: frontend/nobet_app/lib/models/group_membership.dart
class GroupMembership {
  final String userId;
  final String userName;
  final String userEmail;
  final String role;
  final double score;
  final bool isActive;
  final String joinedAt;

  const GroupMembership({
    required this.userId,
    required this.userName,
    required this.userEmail,
    required this.role,
    required this.score,
    required this.isActive,
    required this.joinedAt,
  });

  factory GroupMembership.fromJson(Map<String, dynamic> j) => GroupMembership(
        userId: j['userId'],
        userName: j['userName'] ?? '',
        userEmail: j['userEmail'] ?? '',
        role: j['role'] ?? 'Member',
        score: (j['score'] as num?)?.toDouble() ?? 0.0,
        isActive: j['isActive'] ?? true,
        joinedAt: j['joinedAt'] ?? '',
      );

  bool get isAdmin => role == 'Admin';
}

File: backend/src/NobetSistemi.Domain/Entities/GroupMembership.cs
using NobetSistemi.Domain.Enums;

namespace NobetSistemi.Domain.Entities;

public class GroupMembership
{
    public Guid Id { get; set; }
    public Guid GroupId { get; set; }
    public Group Group { get; set; } = null!;
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public GroupRole Role { get; set; } = GroupRole.Member;
    public double Score { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime JoinedAt { get; set; } = DateTime.UtcNow;
}
